devicetypes/fibargroup/fibaro-walli-double-switch.src/fibaro-walli-double-switch.groovy

```diff
@@ -0,0 +1,533 @@
+		if (includeMeterGet) {
+			cmds << encap(zwave.meterV3.meterGet(scale: 0), it)
+			cmds << encap(zwave.meterV3.meterGet(scale: 2), it)
+		}
+	}
+	delayBetween(cmds, 200)
+}
+
+private resetAll() {
+	childDevices.each { childReset(it.deviceNetworkId) }
+	sendHubCommand reset()
+}
+
+def childReset(deviceNetworkId) {
+	def switchId = getSwitchId(deviceNetworkId)
+	if (switchId != null) {
+		log.debug "Child reset switchId: ${switchId}"
+		sendHubCommand reset(switchId)
+	}
+}
+
+def reset(endpoint = 1) {
+	log.debug "Resetting endpoint: ${endpoint}"
+	delayBetween([
+		encap(zwave.meterV3.meterReset(), endpoint),
+		encap(zwave.meterV3.meterGet(scale: 0), endpoint),
+		"delay 500"
+	], 500)
+}
+
+def getSwitchId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	return (split.length > 1) ? split[1] as Integer : null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		log.debug "encap command: ${cmd} "
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private addChildSwitches(numberOfSwitches) {
+	for (def endpoint : 2..numberOfSwitches) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName + " ${endpoint}"
+			addChildDevice("Child Metering Switch", childDni, device.getHub().getId(), [
+				completedSetup	: true,
+				label			: componentLabel,
+				isComponent		: false
+			])
+		} catch(Exception e) {
+			log.debug "Exception: ${e}"
+		}
+	}
+}
+
+private getParameterMap() {[
+	[
+		name: "LED frame - colour when ON", key: "ledFrame-ColourWhenOn", type: "enum",
+		parameterNumber: 11, size: 1, defaultValue: 1,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta",
+			8: "colour changes smoothly depending on measured power",
+			9: "colour changes in steps depending on measured power"
+		],
+		description: "This parameter defines the LED colour when thedevice is ON. When set to 8 or 9, LED frame colour will change depending on he measured power and parameter 10. Other colours are set permanently and do not depend on power consumption."
+	],
+	[
+		name: "LED frame - colour when OFF", key: "ledFrame-ColourWhenOff", type: "enum",
+		parameterNumber: 12, size: 1, defaultValue: 0,
+		values: [
+			0: "LED disabled",
+			1: "White",
+			2: "Red",
+			3: "Green",
+			4: "Blue",
+			5: "Yellow",
+			6: "Cyan",
+			7: "Magenta"
+		],
+		description: "This parameter defines the LED colour when the device is OFF."
+	],
+	[
+		name: "LED frame - brightness", key: "ledFrame-Brightness", type: "range",
+		parameterNumber: 13, size: 1, defaultValue: 100,
+		range: "1..102",
+		description: "This parameter allows to adjust the LED frame brightness."
```

Comment:
I'd change to:

"adjust the LED frame brightness"

In fact for most of these I'd drop the "this parameter defines/allows" wording

Reply:
ok, it will be fixed (all device settings details were taken - as they were -  from wave alliance official product data)

Reply:
ok, fixed